doc/dune-files.rst

```diff
@@ -1227,6 +1227,9 @@ Dune accepts three kinds of preprocessing:
 - ``(staged_pps <ppx-rewriters-and-flags>)`` is similar to ``(pps ...)``
   but behave slightly differently and is needed for certain ppx rewriters
   (see below for details)
+- ``compat`` is a special value that brings some of the newer OCaml
```

Comment:
How about `compat-syntax`? or `future-syntax`? (that's similar to what python uses)

Reply:
I like `future-syntax`, let's go for that

Reply:
I'm using `future_syntax` to be consistent with the other cases